itch_dl/downloader.py

```diff
@@ -248,6 +249,39 @@ def get_decompressed_content_size(target_path: str | os.PathLike[str]) -> None |
 
         return None
 
+    def map_metadata_to_hh(metadata):
+        tags = []
+        metadata_hh = {}
+        metadata_hh["screenshots"] = screenshot_names
+        metadata_hh["typetag"] = "game"
+        if "links" in metadata["extra"]:
+            for link in metadata["extra"]["links"].keys():
+                if "source" in link.lower():
+                    metadata_hh["repository"] = metadata["extra"]["links"][link]
+        if "license" in metadata["extra"]:
+            for value in metadata["extra"]["license"].keys():
+                if "source" in link.lower():
+                    metadata_hh["gameLicense"] = value
+        if "genre" in metadata["extra"]:
+            for value in metadata["extra"]["genre"].keys():
+                tags.append(value)
```

Comment:
- Links can contain arbitrary text, but the prefilled metadata selection can contain "GitHub" or "Source code", so let's try matching on more keys.
- `metadata["extra"]["license"]` does not seem to ever exist, but `assets_license` and `code_license` are there. Should one or the other be used?
- You can set the license without providing a link to its source code. Do you want to keep the `if source_code_found` check?

```suggestion
        metadata_hh["typetag"] = "game"

        source_code_found = False
        if "links" in metadata["extra"]:
            for link_name, link in metadata["extra"]["links"].items():
                if any(key in link_name.lower() for key in ("github", "source", "repo")):
                    metadata_hh["repository"] = link
                    source_code_found = True

        if "code_license" in metadata["extra"]:
            for value in metadata["extra"]["code_license"].keys():
                if source_code_found:
                    metadata_hh["gameLicense"] = value

        if "genre" in metadata["extra"]:
            tags = list(metadata["extra"]["genre"].keys())
```

Reply:
Done